x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -201,40 +208,57 @@ export class DataRecognizer {
       })
     );
 
-    return configs;
+    const soConfigs = (await this._loadSavedObjectModules()).map((module) => ({
+      dirName: '',
```

Comment:
I find it a bit confusing having an empty string as a `dirName`.  Maybe we should update the `Config` instead, or simply create a dedicated one for SavedObjects configs?

Reply:
https://github.com/elastic/kibana/pull/92855/commits/961396d595104dfe9d633151737a81fde9764366